director/instances.go

```diff
@@ -0,0 +1,25 @@
+package director
+
+import (
+	"strings"
+)
+
+func (d DeploymentImpl) InstanceInfos() ([]VMInfo, error) {
+	infos, err := d.client.DeploymentInstanceInfos(d.name)
+
+	if err != nil && strings.Contains(err.Error(), "404") {
```

Comment:
a bit sketchy to check for "404". i actually think we can just require director to be new enough to have /instances, so i dont think we need to fallback. prolly in future have to introduce first class type of error to handle fallback cases.